### ISSUE TITLE
Is there a way to obtain the trace back matrix in align_pairwiseAlignment.c?

### ISSUE THREAD

**Issue by peteryzheng** [Bioconductor/Biostrings#108]
Hi, 

TL;DR: Would it be possible to have `pairwiseAlignment` return the traceback matrix?

I am interested in doing global-local alignments between pattern string A and a set of subject strings B, C, D, etc. In this case, the subject strings are longer than the pattern string. I hope to find within the subject strings those that have significantly better alignment scores than we would expect by chance. 

Furthermore, I want to adjust for how far from the start of the subject string where a good alignment is found. This is biologically important in my use case. The reason is that since my pattern strings tend to be short, for instance, if we try to find a 6-mer in a set of 100-mers, chances are we are going to find some hits. However, if we find a 6-mer right at the start of a certain 100-mer, that would indicate something significant biologically in our problem.  

In order to control for this, I am doing iteratively global-local alignments using `pairwiseAlignment` between pattern string A and all substrings of subject strings that start from the beginning. [for instance, from `substring(B, 1, nchar(A))` to `substring(B, 1, nchar(B))`]. Then we use those alignment scores that are controlled for distance from the starting point to assess significance. A major problem with this current approach is that we have to run the underlying alignment algorithm many many times for each substring. 

However, If we were to have access to the traceback matrix `currTraceMatrix` in [align_pairwiseAlignment.c](https://github.com/Bioconductor/Biostrings/blob/d13154e5013363a568d983ff26e483e6c77fb007/src/align_pairwiseAlignment.c#L570-L571) from the `pairwiseAlignment` call, that would save me a tremendous amount of time. 

Would this be possible? I would imagine some folks in my field would be interested in having this feature. 

Thank you in advance. 

Best,
Peter
